redash/models/parameterized_query.py

```diff
@@ -122,23 +131,32 @@ def __init__(self, template, schema=None, org=None):
         self.parameters = {}
 
     def apply(self, parameters):
-        invalid_parameter_names = [key for (key, value) in parameters.items() if not self._valid(key, value)]
-        if invalid_parameter_names:
-            raise InvalidParameterError(invalid_parameter_names)
+        # filter out params not defined in schema
+        if self.schema:
+            names_with_definition = lpluck("name", self.schema)
+            parameters = {k: v for (k, v) in parameters.items() if k in names_with_definition}
+
+        invalid_parameters = compact({k: self._invalid_message(k, v) for (k, v) in parameters.items()})
+        if invalid_parameters:
+            raise InvalidParameterError(invalid_parameters)
         else:
             self.parameters.update(parameters)
             self.query = mustache_render(self.template, join_parameter_list_values(parameters, self.schema))
 
         return self
 
-    def _valid(self, name, value):
+    def _invalid_message(self, name, value):
+        if value is None:
+            return 'Required parameter'
+
+        # skip if no schema
         if not self.schema:
-            return True
+            return None
 
         definition = next((definition for definition in self.schema if definition["name"] == name), None)
 
         if not definition:
-            return False
+            return 'Parameter no longer exists in query.'
```

Comment:
Kept as safety but can not be reached since no-definition params are filtered out beforehand.